hecuba_py/src/hdict.py

```diff
@@ -529,6 +536,21 @@ def __getitem__(self, key):
             else:
                 return final_results[0]
 
+    def __make_val_persistent(self, val, col=0):
+        if isinstance(val, StorageDict):
+            for k, element in val.iteritems():
+                val[k] = self.__make_val_persistent(element)
+        elif isinstance(val, list):
+            for index, element in enumerate(val):
+                val[index] = self.__make_val_persistent(element, index)
+        if isinstance(val, IStorage) and not val._is_persistent:
+            val._storage_id = uuid.uuid4()
+            attribute = self._columns[col][0]
+            count = self._count_name_collision(attribute)
+            # new name as ksp+table+obj_class_name
+            val.make_persistent(self._ksp + '.' + self._table + "_" + attribute + "_" + str(count))
```

Comment:
I think that the name should be:
if count==0
name = self._ksp + '.' + self._table + "_" + attribute
else
name = self._ksp + '.' + self._table + "_" + attribute + "_" + str(count - 1t)